pyro/distributions/torch_patch.py

```diff
@@ -45,4 +47,51 @@ def _torch_dirichlet_grad(x, concentration, total):
     return unpatched_fn(x, concentration, total)
 
 
+# This version of broadcast_all() is compatible with early versions of the PyTorch jit,
+# since it avoids torch._C._infer_size(). However it is more expensive since it infers
+# size by summing the tensors. It is mainly useful for working around one jit limitation
+# to discovering additional jit limitations.
+#
+# To temporarily apply this patch, uncomment one or more of the decorators:
+#
+# @_patch('torch.distributions.beta.broadcast_all')
+# @_patch('torch.distributions.dirichlet.broadcast_all')
+# @_patch('torch.distributions.normal.broadcast_all')
+# @_patch('torch.distributions.utils.broadcast_all')
+def _broadcast_all(*values):
+    r"""
+    Given a list of values (possibly containing numbers), returns a list where each
+    value is broadcasted based on the following rules:
+      - `torch.*Tensor` instances are broadcasted as per the `broadcasting rules
+        <http://pytorch.org/docs/master/notes/broadcasting.html>`_
+      - numbers.Number instances (scalars) are upcast to tensors having
+        the same size and type as the first tensor passed to `values`.  If all the
+        values are scalars, then they are upcasted to Tensors having size
+        `(1,)`.
+
+    Args:
+        values (list of `numbers.Number` or `torch.*Tensor`)
+
+    Raises:
+        ValueError: if any of the values is not a `numbers.Number` or
+            `torch.*Tensor` instance
+    """
+    values = list(values)
+    scalar_idxs = [i for i in range(len(values)) if isinstance(values[i], Number)]
+    tensor_idxs = [i for i in range(len(values)) if values[i].__class__.__name__ == 'Tensor']
+    if len(scalar_idxs) + len(tensor_idxs) != len(values):
+        raise ValueError('Input arguments must all be instances of numbers.Number or torch.tensor.')
+    if tensor_idxs:
+        broadcast_shape = sum(values).size()  # expensive alternative to torch._C._infer_size()
```

Comment:
Calling `.size` ends up returning a tuple of pytorch scalars when JITed. I wonder why this gets triggered only in `TraceEnum_ELBO` though.